cmd/runtimetest/main.go

```diff
@@ -370,6 +371,52 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func getActualCgroupPath() (string, error) {
+	f, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return "", err
+		}
+
+		text := s.Text()
+		parts := strings.Split(text, ":")
+		return parts[2], nil
+	}
+	return "", fmt.Errorf("can not get cgroup path")
+}
+
+func validateCgroupsPath(spec *rspec.Spec) error {
+	logrus.Debugf("validating cgroupsPath")
+	expectedPath := spec.Linux.CgroupsPath
+	if expectedPath == nil {
+		return nil
+	}
+	*expectedPath = strings.Replace(*expectedPath, ":", "/", -1)
+
+	actualPath, err := getActualCgroupPath()
+	if err != nil {
+		return err
+	}
+
+	if filepath.IsAbs(*expectedPath) {
+		if *expectedPath != actualPath {
+			return fmt.Errorf("Cgroup path expected: %v, actual: %v", *expectedPath, actualPath)
+		}
+	} else {
+		*expectedPath = strings.Replace(*expectedPath, "../", "", -1)
+		if !strings.Contains(actualPath, *expectedPath) {
```

Comment:
`strings.Contains` is a difficult way to do this (e.g. what if `actualPath` is `ab/cd` but `*expectedPath` is `b/cd`?).  Can we use [`filepath.Rel`][1] instead?

[1]: https://golang.org/pkg/path/filepath/#Rel